Add all 5 pages of Chapter 1

annekainicUSDS opened the pull request:
Adding all 5 pages of Chapter 1. Connected to https://github.com/department-of-veterans-affairs/vets.gov-team/issues/2230.

### Veteran Information
<img width="686" alt="screen shot 2017-04-20 at 2 36 10 pm" src="https://cloud.githubusercontent.com/assets/25183456/25246799/c500d304-25d6-11e7-81c8-b1af2acde397.png">

### Birth information
<img width="784" alt="screen shot 2017-05-01 at 1 12 11 pm" src="https://cloud.githubusercontent.com/assets/25183456/25586978/e64d71c4-2e6f-11e7-9cc7-2763b52616df.png">

Pages 3-5 added here: https://github.com/department-of-veterans-affairs/vets-website/pull/5463

### Questions
1. When navigating to each page, the first element on the page is styled as active. I can't figure out why.
<img width="509" alt="screen shot 2017-04-20 at 2 31 56 pm" src="https://cloud.githubusercontent.com/assets/25183456/25246612/33c446a0-25d6-11e7-910e-9da2d76e31c8.png">

2. I tried applying a custom error message to first name (see code), but it isn't appearing. Can't figure out why.
<img width="499" alt="screen shot 2017-04-20 at 2 33 02 pm" src="https://cloud.githubusercontent.com/assets/25183456/25246644/4ca04318-25d6-11e7-817c-36fec550da95.png">

3. I'm keeping the paths as they currently are. We probably want to clean them up though. Related to this ticket: https://github.com/department-of-veterans-affairs/vets.gov-team/issues/1746

4. In order to get this to work, I had to amend the schema in a few places. Permanent fixes to the schema are in this PR: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/87

5. On the birthInformation page, I couldn't figure out how to add another title of "Place of Birth" in the middle of the page, before `City of Birth` and `State of Birth`. Looks like it might involve creating the `uiSchema` in a separate file and adding a `ui:title` there (adding it directly in `config.js` just added it at the top of the page.

annekainicUSDS committed:
Add some tests

annekainicUSDS commented on src/js/hca-rjsf/config/form.js:
```diff
@@ -17,11 +32,86 @@ const formConfig = {
+            veteranFullName: _.merge(fullNameUISchema, {
+              first: {
+                'ui:errorMessages': {
+                  minLength: 'Please provide a valid name. Must be at least 2 characters.'
```
This isn't actually rendering

jbalboni replied:
It looks like for the first name, the minLength is 1. The last name is 2. If I put this on the `last` field, it shows up.

annekainicUSDS commented on src/js/hca-rjsf/config/form.js:
```diff
@@ -17,11 +32,86 @@ const formConfig = {
+                }
+              }),
+              mothersMaidenName: {
+                type: 'string'
```
We can remove this once this PR is merged: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/87

annekainicUSDS commented on src/js/hca-rjsf/config/form.js:
```diff
@@ -17,11 +32,86 @@ const formConfig = {
+              veteranFullName: _.merge(fullName, {
+                properties: {
+                  suffix: {
+                    type: 'string'
```
We can remove this once this PR is merged: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/87

annekainicUSDS commented on src/js/hca-rjsf/config/form.js:
```diff
@@ -17,11 +32,86 @@ const formConfig = {
+              veteranSocialSecurityNumber: ssn.schema,
+              cityOfBirth,
+              stateOfBirth: _.merge(stateOfBirth, {
+                type: 'string'
```
We can remove this once this PR is merged: https://github.com/department-of-veterans-affairs/vets-json-schema/pull/87

jbalboni commented:
@annekainicUSDS For 1, I think it's this rule in edu-benefits.scss (line 225ish):

```
.va-nav-breadcrumbs-list, legend, .nav-header, .edu-page-title, .usa-input-error, .input-error-date {
  &:focus {
    outline: none;
  }
}
```

We should probably move most of that to a common scss file.

jbalboni commented:
@annekainicUSDS For the place of birth title, I've been creating a `view` object with a title. So like this in the uiSchema:

```js
'view:placeOfBirth': {
  'ui:title': 'Place of birth',
  cityOfBirth: {
    'ui:title': 'City of birth'
  },
  stateOfBirth: {
    'ui:title': 'State of birth'
  }
}
```

And the properties object in the regular schema: 

```js
properties: {
  veteranDateOfBirth: date,
  veteranSocialSecurityNumber: ssn.schema,
  'view:placeOfBirth': {  
    type: 'object',
    properties: { 
      cityOfBirth,
      stateOfBirth: _.merge(stateOfBirth, {
        type: 'string'
      })
    }
  }
}
```

jbalboni reviewed (commented):
This looks good so far to me.

annekainicUSDS committed:
Include edu benefits scss in hca for now, long term solution to move common styles to the module for schemaform styles

annekainicUSDS committed:
Merge in latest from master

annekainicUSDS commented on src/js/hca-rjsf/hca-rjsf-entry.jsx:
```diff
@@ -12,6 +12,7 @@ import reducer from './reducer';
 
 require('../common');
 require('../../sass/hca.scss');
+require('../../sass/edu-benefits.scss');
```
For now I'm just going to include the edu-benefits scss file. I started parsing out what was in this file to make it more generic for both edu and hca, but it started to turn into a big project and figured it was better to tackle it in a new PR.

annekainicUSDS commented:
Addressed all comments and fixed the issues I identified in the description above (thanks to @jbalboni for the solutions to those). Should be able to undergo another review now.

annekainicUSDS committed:
Instead of populating stateOfBirth field with values from schema, use front-end values and filter for US states

annekainicUSDS committed:
Add demographic page

annekainicUSDS committed:
Contact information page working

annekainicUSDS committed:
Add tests for demographic page and contact info page

annekainicUSDS committed:
Veteran address page

annekainicUSDS committed:
Update react test library

annekainicUSDS committed:
Fix code climate issues

jbalboni commented on src/js/hca-rjsf/config/form.js:
```diff
@@ -69,9 +73,94 @@ const formConfig = {
+                  stateOfBirth: {
+                    type: 'string',
+                    'enum': states.USA.map(state => state.value),
+                    enumNames: states.USA.map(state => state.label)
```
It doesn't save you that much, I guess, but this file has a `stateLabels` object that you could use in `ui:options.labels`. See the `vaFacility` config for an example. 

That would remove the need for `enumNames` and save the extra iteration of the states list.

annekainicUSDS replied:
Yep, that makes sense, I can make that change

jbalboni commented on test/hca-rjsf/config/veteranInformation.unit.spec.jsx:
```diff
@@ -0,0 +1,37 @@
+import formConfig from '../../../src/js/hca-rjsf/config/form.js';
+
+describe('HCA veteranInformation', () => {
+  it('should render veteranInformation page', () => {
```
Same comment as on the other PR, I think we want some tests around submitting the form and checking that the right number of required errors happen.

annekainicUSDS replied:
Yep, going to add them in the other PR because it's easier to do in one PR.

jbalboni reviewed (approved):
Couple of minor things, but looks good.

annekainicUSDS committed:
Add tests for required fields for all pages

annekainicUSDS committed:
Make updates from review

annekainicUSDS committed:
Use stateLabels for stateOfBirth

annekainicUSDS committed:
Remove TODO because code climate failure bothers me

annekainicUSDS committed:
Update vets-json-schema version

annekainicUSDS committed:
Remove unnecessary type strings
